src/modules/Dropdown/Dropdown.js

```diff
@@ -1,111 +1,97 @@
-      PropTypes.bool,
+      PropTypes.number,
     ]),
-    saveRemoteData: PropTypes.bool,
-    useLabels: PropTypes.bool,
-
-    // Additional Settings
-    direction: PropTypes.string,
-    keepOnScreen: PropTypes.bool,
-    context: PropTypes.object,
-    fullTextSearch: PropTypes.bool,
-    preserveHTML: PropTypes.bool,
-    sortSelect: PropTypes.bool,
-    showOnFocus: PropTypes.bool,
-    allowTab: PropTypes.bool,
-    transition: PropTypes.string,
-    duration: PropTypes.number,
-    keys: PropTypes.object,
-    delay: PropTypes.object,
-
-    // Callbacks
-    onChange: PropTypes.func,
-    onShow: PropTypes.func,
-    onHide: PropTypes.func,
-    onNoResults: PropTypes.func,
-    onLabelSelect: PropTypes.func,
-    onLabelRemove: PropTypes.func,
-    onLabelCreate: PropTypes.func,
-    onAdd: PropTypes.func,
-    onRemove: PropTypes.func,
+    options: PropTypes.arrayOf(PropTypes.shape({
+      value: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]),
+      text: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]),
+    })),
+  };
+
+  static defaultProps = {
+    icon: 'dropdown',
   };
 
   componentDidMount() {
-    this.element = $(this.refs.select)
-    this.element.dropdown({
-      // Settings
-      action: this.props.action,
-      allowAdditions: this.props.allowAdditions,
-      allowCategorySelection: this.props.allowCategorySelection,
-      apiSettings: this.props.apiSettings,
-      fields: this.props.fields,
-      forceSelection: this.props.forceSelection,
-      glyphWidth: this.props.glyphWidth,
-      label: this.props.label,
-      match: this.props.match,
-      maxSelections: this.props.maxSelections,
-      on: this.props.on,
-      placeholder: this.props.placeholder,
-      saveRemoveData: this.props.saveRemoteData,
-      useLabels: this.props.useLabels,
-
-      // Additional Settings
-      direction: this.props.delay,
-      keepOnScreen: this.props.keys,
-      context: this.props.duration,
-      fullTextSearch: this.props.transition,
-      preserveHTML: this.props.allowTab,
-      sortSelect: this.props.showOnFocus,
-      showOnFocus: this.props.sortSelect,
-      allowTab: this.props.preserveHTML,
-      transition: this.props.fullTextSearch,
-      duration: this.props.context,
-      keys: this.props.keepOnScreen,
-      delay: this.props.direction,
-
-      // Callbacks
-      onChange: this.props.onChange,
-      onHide: this.props.onHide,
-      onShow: this.props.onShow,
-      onNoResults: this.props.onNoResults,
-      onLabelSelect: this.props.onLabelSelect,
-      onLabelRemove: this.props.onLabelRemove,
-      onLabelCreate: this.props.onLabelCreate,
-      onAdd: this.props.onAdd,
-      onRemove: this.props.onRemove,
-    })
+    this.refresh()
   }
 
   componentDidUpdate(prevProps, prevState) {
-    this.element.dropdown('refresh')
+    if (prevProps === this.props) {
+      this.refresh()
+    }
```

Comment:
When the component updates, wouldn't we want to refresh only if `props` did change? And, if so, wouldn't we do that with a shallow or deep compare given the `options` and `children` props are objects themselves?